tapchannel/aux_leaf_signer_test.go

```diff
@@ -0,0 +1,172 @@
+package tapchannel
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/ecdsa"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/lightninglabs/taproot-assets/asset"
+	cmsg "github.com/lightninglabs/taproot-assets/tapchannelmsg"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/tlv"
+	"github.com/stretchr/testify/require"
+)
+
+// Some of these test values and functions are from lnd's lnwire/lnwire_test.go.
+var (
+	testRBytes, _ = hex.DecodeString(
+		"8ce2bc69281ce27da07e6683571319d18e949ddfa2965fb6caa1bf0314f8" +
+			"82d7",
+	)
+	testSBytes, _ = hex.DecodeString(
+		"299105481d63e0f4bc2a88121167221b6700d72a0ead154c03be696a292d" +
+			"24ae",
+	)
+	testRScalar = new(btcec.ModNScalar)
+	testSScalar = new(btcec.ModNScalar)
+	_           = testRScalar.SetByteSlice(testRBytes)
+	_           = testSScalar.SetByteSlice(testSBytes)
+	testSig     = ecdsa.NewSignature(testRScalar, testSScalar)
+
+	testSchnorrSigStr, _ = hex.DecodeString(
+		"04e7f9037658a92afeb4f25bae5339e3ddca81a353493827d26f16d92308" +
+			"e49e2a25e92208678a2df86970da91b03a8af8815a8a60498b35" +
+			"8daf560b347aa557",
+	)
+	testSchnorrSig, _ = lnwire.NewSigFromSchnorrRawSignature(
+		testSchnorrSigStr,
+	)
+
+	startNumHTLCs = 5
+	endNumHTLCs   = input.MaxHTLCNumber
+)
```

Comment:
I recommend adding comments to explain the purpose of these fields. For instance, what does `testRBytes` represent?

Reply:
Does `startNumHTLCs` really need a global scope? I think it's only used in `TestMaxCommitSigMsgSize`.